pkg/nats/transport/nats.go

```diff
@@ -209,6 +265,17 @@ func CreateClient(ctx context.Context, config *NATSTransportConfig) (*nats_helpe
 		nats.Name(config.NodeID),
 		nats.MaxReconnects(-1),
 	}
+
+	// We need to do this logic since the Nats Transport Layer does not differentiate
+	// between orchestrator mode and compute mode
+	if config.ServerTLSCert == "" && config.ClientTLSCACert != "" {
+		// this client is for a compute node
+		clientOptions = append(clientOptions, nats.RootCAs(config.ClientTLSCACert))
+	} else if config.ServerTLSCert != "" && config.ServerTLSCACert != "" {
+		// this client is for a orchestrator node
+		clientOptions = append(clientOptions, nats.RootCAs(config.ServerTLSCACert))
+	}
+
```

Comment:
this is difficult to follow. Should the transport accept a single ca cert and let the caller (ie `pk/node/Node`) pass the right one? 

Still things will be tricky when you run a hybrid node. I guess in that mode we will get whatever CA is provided until we refactor that whole logic, or maybe refactor it with this PR

Reply:
I agree with you. This whole logic was added since the Nats Transport Layer does not differentiate between orchestrator mode and compute mode, and I wanted (without refactoring the code) to explicitly set the boundary. 

I know it does not look pretty, but it is a compromise when compared with refactoring the whole NATS transport